optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/bi/BavetFilterBiConstraintStream.java

```diff
@@ -51,8 +51,8 @@ public BavetFilterBiConstraintStream(BavetConstraintFactory<Solution_> constrain
 
     @Override
     protected BavetFilterBiNode<A, B> createNode(BavetNodeBuildPolicy<Solution_> buildPolicy,
-            Score<?> constraintWeight, int nodeOrder, BavetAbstractBiNode<A, B> parentNode) {
-        return new BavetFilterBiNode<>(buildPolicy.getSession(), nodeOrder, parentNode, predicate);
+            Score<?> constraintWeight, BavetAbstractBiNode<A, B> parentNode) {
+        return new BavetFilterBiNode<>(buildPolicy.getSession(), buildPolicy.getNodeIndexMaximum() + 1, parentNode, predicate);
```

Comment:
(important)
This is very wierd.
I would expect the it take as the next Index from the buildPolicy.
This way the next Index doesn't increment. That's a bug I think?

Reply:
The index is incremented immediately after this, when the node is checked for sharing. There is a test that checks for this.
I am interested in counter-examples.